core/controllers/question_editor.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the questions editor, from where questions are edited
+and are created.
+"""
+import logging
+
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import question_domain
+from core.domain import question_services
+import feconf
+import utils
+
+
+class QuestionEditorPage(base.BaseHandler):
+    """The editor page for a single question."""
+
+    @acl_decorators.can_view_any_question_editor
+    def get(self, question_id):
+        """Handles GET requests."""
+
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question = question_services.get_question_by_id(question_id)
+
+        if question is None:
+            raise self.PageNotFoundException(
+                'The question with the given id doesn\'t exist.')
+
+        self.values.update({
+            'question_id': question.to_dict(),
+            'can_edit': question_services.check_can_edit_question(
+                self.user.user_id, question.question_id)
```

Comment:
This is just self.user_id, and isn't it question.id?

Reply:
I replicated as for explorations to self.user.user_id. And changed